### PR TITLE
Have consumer spans name follow service name attribute

### PR DESCRIPTION
**Pull request by atshaw43**
**Description:** <Describe what has changed.>
Kafka consumer spans are not leveraging the service name for their span names.

**Link to tracking Issue:**
https://github.com/aws-observability/aws-otel-collector/issues/894

**Testing:**
Unit tests

### CODE REVIEW

File: exporter/awsxrayexporter/internal/translator/segment.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func MakeSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []str
 		}
 	}
 
-	if name == "" && span.Kind() == ptrace.SpanKindServer {
+	if name == "" && (span.Kind() == ptrace.SpanKindServer || span.Kind() == ptrace.SpanKindConsumer) {
 		// Only for a server span, we can use the resource.
```
Comment (bryan-aguilar) [inline exporter/awsxrayexporter/internal/translator/segment.go:191]: This comment should probably be updated also

File: exporter/awsxrayexporter/internal/translator/segment_test.go [hunk 2]
```diff
@@ -1103,6 +1103,47 @@ func TestServerSpanWithAwsLocalServiceName(t *testing.T) {
...
+	parentSpanID := newSegmentID()
+	attributes := make(map[string]any)
+	attributes[conventions.AttributeHTTPMethod] = "POST"
+	attributes[conventions.AttributeHTTPScheme] = "https"
+	attributes[conventions.AttributeHTTPHost] = "payment.amazonaws.com"
+	attributes[conventions.AttributeHTTPTarget] = "/"
+	attributes[conventions.AttributeRPCService] = "ABC"
+
+	resource := constructDefaultResource()
+	span := constructServerSpan(parentSpanID, spanName, 0, "OK", attributes)
```
Comment (bryan-aguilar) [inline exporter/awsxrayexporter/internal/translator/segment_test.go:1117]: You are constructing a server span here and not a consumer span. The unit test should be updated to reflect the functionality under test.

**Review by bryan-aguilar — CHANGES_REQUESTED** [review]

**Review by bryan-aguilar — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Comment by evan-bradley** [comment]
@wangzlei @srprash could you please take a look as code owners?

**Comment by evan-bradley** [comment]
@wangzlei @srprash Please take a look.

**Pull request closed without merging**